### DISCUSSION TITLE
Set search.max_buckets for ES

### DISCUSSION THREAD

**Discussion by ben-sec** [Security-Onion-Solutions/securityonion#6893] [Unsupported Versions] [answered]
Hello!

I have to set a custom search.max_buckets for Elasticsearch and was wondering if I can set this in `/opt/so/saltstack/local/pillar/global.sls` on my manager node for my distributed deployment? One of my Elastalert rules is crashing, because I currently reach the default bucket limit. Thanks in advance!

Cheers, Ben

**Comment by weslambert** [comment, marked as answer]
You should be able to add it to the global, but you might also consider adjusting per node in the minion pillar file(s), under the `elasticsearch.config` section.

Are you utilizing custom queries, or complex/intensive visualizations in Kibana? Or is this caused when performing groupbys in SOC? I'm curious how you reached the limit, as many folks don't seem to have this issue. Could it be related to the cardinality and range of the data across the search timeframe? How is this error produced?

**Reply by ben-sec** [reply]
I'm using a new_term Elastalert to identify new DNS querries by watching the field dns.query.name. When my `terms_window_size` is greater than 25 I run into this bucket limit.

**Reply by weslambert** [reply]
What is your average daily log volume (in events/GB/TB), and how far back are you searching with the rule?

**Reply by ben-sec** [reply]
We are talking about 25 million daily zeek logs and the Elastalert rule is checking the last 25 days. More than 25 days is crashing. The overall daily log volume is about 50 million logs (incl. firewall logs, Windows event logs and the rest of SO).

**Reply by weslambert** [reply]
Understood -- again, Elastic states that changing the value for this setting can be detrimental to cluster performance if you are not careful. You might try it and see if it works for you.  Otherwise, I would try backing off the timeframe and frequency of the Elastalert rule, and maybe what ever open indices you have in your cluster(s).

**Reply by ben-sec** [reply]
OK, I will stick with my 25 days.
